### PR TITLE
Replace `thrust/std::get` with structure bindings

### PR DESCRIPTION
**Pull request by codereport**
While watching @bdice and @isVoid's Better Code talk, I noticed a pair of `thrust::get`s that could be replaced by structured bindings. I did a quick search and found all the places the clean up could be done. That is what this PR does.

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

File: cpp/src/merge/merge.cu [hunk 1]
```diff
@@ -80,9 +80,8 @@ __global__ void materialize_merged_bitmask_kernel(
   auto active_threads = __ballot_sync(0xffffffff, destination_row < num_destination_rows);
 
   while (destination_row < num_destination_rows) {
-    index_type const& merged_idx = merged_indices[destination_row];
-    side const src_side          = thrust::get<0>(merged_idx);
-    size_type const src_row      = thrust::get<1>(merged_idx);
+    index_type const& merged_idx   = merged_indices[destination_row];
+    auto const [src_side, src_row] = merged_idx;
```
Comment (hyperbolic2346) [inline cpp/src/merge/merge.cu:84]:
```suggestion
    auto const [src_side, src_row] = merged_indices[destination_row];
```

`merged_idx` doesn't seem to be used anywhere else, why have it at all?

**Review by hyperbolic2346 — APPROVED** [review]
Small nit. Thanks for doing this!

File: cpp/src/merge/merge.cu (hunk 1 above)
Comment (codereport) [reply cpp/src/merge/merge.cu]: Done :+1:

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**